app/models/api_response.rb

```diff
@@ -0,0 +1,23 @@
+class ApiResponse
+  attr_reader :code
+  def initialize(response)
+    @code = response.code.to_i
+    @response_body = response.body
+  end
+
+  def succeeded?
+    code == 200
+  end
+
+  def failed?
+    !succeeded?
+  end
+
+  def body
+    failed? ? {} : JSON.parse(@response_body).with_indifferent_access
```

Comment:
[suggestion] Maybe consider reversing this to be `succeeded? ? JSON.parse(...) : {}` ?